The flag --name has been renamed to --override-name in newer auroraboot

jimmykarily opened the pull request.

jimmykarily committed:
The flag --name has been renamed to --override-name in newer auroraboot

Signed-off-by: Dimitris Karakasilis <dimitris@karakasilis.me>

Itxaka commented:
This will only work if aurora has the 0.14 or later version rigth? For older ones it would still fail?

Itxaka reviewed (approved):
this unblocks the latest but I still think that each osbuilder should have a fixed aurora version, so we dont encounter more issues like this downthe line. Then we would just update the aurora version in the code and fix the flags if needed

jimmidyson commented:
> osbuilder should have a fixed aurora version

Agreed! Also would be good to run e2e tests nightly so we can get faster feedback that things are broken if dependency projects change.

jimmykarily committed:
Set the default tool-image to a compatible one

We'd better set this to the compatible auroraboot version each time
because changing the flags of auroraboot in the future means it may not always
work with `latest`.

Signed-off-by: Dimitris Karakasilis <dimitris@karakasilis.me>

jimmykarily commented:
> this unblocks the latest but I still think that each osbuilder should have a fixed aurora version, so we dont encounter more issues like this downthe line. Then we would just update the aurora version in the code and fix the flags if needed

Good point. I think this covers it: https://github.com/kairos-io/osbuilder/pull/200/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R60

jimmykarily commented:
> > osbuilder should have a fixed aurora version
> 
> Agreed! Also would be good to run e2e tests nightly so we can get faster feedback that things are broken if dependency projects change.

Now that the auroraboot version will be stubbed, there is no way to break it without explicitly changing the version or the code. That means, whoever changes it will see it on the PR. No reason to run nightly then?

jimmykarily commented:
I run the e2e tests locally (`make kind-e2e-tests`) and I see this error in the test cluster:

```
 + auroraboot --debug build-iso --override-name simple-m585t --date=false --output /artifacts dir:/root │
│ 2025-11-20T07:56:42Z INF [1] Copying /rootfs source to /artifacts/temp-rootfs                          │
│ 2025-11-20T07:56:42Z INF [1] Starting rsync...                                                         │
│ 2025-11-20T07:56:42Z INF Copying cloud config cloudConfig=                                             │
│ 2025-11-20T07:56:43Z INF [1] Finished syncing                                                          │
│ 2025-11-20T07:56:43Z INF [1] Finished copying /rootfs into /artifacts/temp-rootfs                      │
│ 2025-11-20T07:56:43Z INF Generating iso 'simple-m585t' from '/artifacts/temp-rootfs' to '/artifacts'   │
│ 2025-11-20T07:56:43Z INF [1] Preparing squashfs root...                                                │
│ 2025-11-20T07:56:43Z INF [1] Copying /artifacts/temp-rootfs source to /tmp/auroraboot-iso063937313/roo │
│ 2025-11-20T07:56:43Z INF [1] Starting rsync...                                                         │
│ 2025-11-20T07:56:45Z INF [1] Finished syncing                                                          │
│ 2025-11-20T07:56:45Z INF [1] Finished copying /artifacts/temp-rootfs into /tmp/auroraboot-iso063937313 │
│ 2025-11-20T07:56:45Z INF [1] Preparing ISO image root tree...                                          │
│ 2025-11-20T07:56:45Z INF [1] Copying /tmp/geniso126679816 source to /tmp/auroraboot-iso063937313/iso   │
│ 2025-11-20T07:56:45Z INF [1] Starting rsync...                                                         │
│ 2025-11-20T07:56:45Z INF [1] Finished syncing                                                          │
│ 2025-11-20T07:56:45Z INF [1] Finished copying /tmp/geniso126679816 into /tmp/auroraboot-iso063937313/i │
│ 2025-11-20T07:56:45Z INF [1] Creating EFI image...                                                     │
│ 2025-11-20T07:56:45Z ERR [1] Failed preparing ISO's root tree: KAIROS_ARCH/KAIROS_TARGETARCH not found │
│ 2025-11-20T07:56:45Z ERR Failed generating iso 'simple-m585t' from '/artifacts/temp-rootfs'. Error: 1  │
│     * KAIROS_ARCH/KAIROS_TARGETARCH not found in etc/os-release                                        │
│                                                                                                        │
│                                                                                                        │
│ 1 error occurred:                                                                                      │
│     * KAIROS_ARCH/KAIROS_TARGETARCH not found in etc/os-release                                        │
│                                                                      
```

jimmykarily commented:
Maybe using this image from 2023 doesn't help :D : https://github.com/kairos-io/osbuilder/blob/f3ee5c3dc12387fd225747c9724b002de686ea39/tests/e2e/e2e_simple_test.go#L49

jimmykarily committed:
Use newer image to fix tests

Signed-off-by: Dimitris Karakasilis <dimitris@karakasilis.me>
